smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java

```diff
@@ -1063,6 +1064,52 @@ public void invite(MessageBuilder messageBuilder, EntityBareJid user, String rea
         connection.sendStanza(message);
     }
 
+    /**
+     * Invites another user to the room in which one is an occupant. In contrast
+     * to the method "invite", the invitation is sent directly to the user rather
+     * than via the chat room.  This is useful when the user being invited is
+     * offline, as otherwise the invitation would be dropped.
+     *
+     * @param address the user to send the invitation to
+     * @throws NotConnectedException if the XMPP connection is not connected.
+     * @throws InterruptedException if the calling thread was interrupted.
+     */
+    public void inviteDirectly(EntityBareJid address) throws NotConnectedException, InterruptedException {
+        inviteDirectly(address, null, null, false, null);
+    }
+
+     /**
+     * Invites another user to the room in which one is an occupant. In contrast
+     * to the method "invite", the invitation is sent directly to the user rather
+     * than via the chat room.  This is useful when the user being invited is
+     * offline, as otherwise the invitation would be dropped.
+     *
+     * @param address the user to send the invitation to
+     * @param reason the purpose for the invitation
+     * @param password specifies a password needed for entry
+     * @param continue_bool specifies if the groupchat room continues a one-to-one chat having the designated thread
+     * @param thread the thread to continue
+     * @throws NotConnectedException if the XMPP connection is not connected.
+     * @throws InterruptedException if the calling thread was interrupted.
+     */
+    public void inviteDirectly(EntityBareJid address, String reason, String password, boolean continue_bool, String thread)
+        throws NotConnectedException, InterruptedException {
+        MessageBuilder messageBuilder = connection.getStanzaFactory().buildMessageStanza();
```

Comment:
Use the builder pattern if it is available
```suggestion
        Message message = connection.getStanzaFactory()
            .buildMessageStanza()
            .to(address)
            .addExtension(invitation)
            .build();
```